pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -91,6 +91,7 @@ in rec {
 
     , ... } @ attrs:
 
+      assert !lib.hasAttr "name" attrs -> lib.assertMsg (lib.hasAttrs [ "pname" "version" ] attrs ) "Derivation needs a pname and a version";
```

Comment:
There are like a gazillion derivations in nixpkgs that do not use `pname` and `version`, many of them because they call `mkDerivation` through some wrapper.  This will never work and nixpkgs will just cease to evaluate altogether.

Reply:
I've checked and I couldn't find any derivation that both has a name or a pname and a version.

Reply:
I think the ofborg evaluation error is pretty unambiguous proof that this is not the case.

Reply:
What should work on the other hand is amending this logic to assert when one of `pname` and `version` is unset.

https://github.com/NixOS/nixpkgs/blob/2210eb352bca997658dd3d7221ec6f3f7b8db457/pkgs/stdenv/generic/make-derivation.nix#L191-L192

Reply:
Fixed ? https://github.com/NixOS/nixpkgs/blob/b239ace3264d5ed92ab13dd45087db3345da8e6e/pkgs/stdenv/generic/make-derivation.nix#L192

Reply:
That looks much better semantically, although I would really question the usefulness of `lib.hasAttrs [ "pname" "version" ] attrs` in contrast to `attrs ? pname && attrs ? version`, but these things are not up to me to decide.  I've removed my :-1:.

Reply:
To be honest, the only reason why I made this function is that I didn't knew the ? operator. What's strange is that I thought this was only used to give a default value to an argument in the argument set. It's removed.

Reply:
Then they need a fix or should use name. This won't get merged as long as it has eval errors so there is no chance it will break eval and no reason to thumb it down.